### PR TITLE
IAM docs: Transform `API keys` to `Migrate API keys` docs

### PR DESCRIPTION
**Pull request by ppcano**
Transform `API keys` docs to `Migrate API keys` according to https://github.com/grafana/website/pull/21369#issuecomment-2306809602 (cc @eleijonmarck )

- move the page and create redirect
- minor updates to the intro and headings

Screenshot 2024-08-23 at 21 04 04

This PR is ready for review and merge.

We have also updated the links in  the Swagger/OpenAPI docs and one help button in Grafana:

- https://github.com/grafana/grafana/blob/main/pkg/api/apikey.go#L114

- https://github.com/grafana/grafana/blob/main/public/app/features/api-keys/MigrateToServiceAccountsCard.tsx#L13

Further considerations:

- Mount `Migrate API keys` docs in Grafana Cloud https://github.com/grafana/website/pull/21329
- Convert the section `Migrate Cloud Stack API keys using Terraform` to its own docs page in Grafana Cloud

### CODE REVIEW

**Review by imatwawana — APPROVED** [review]
I'm only codeowner on one file (Breaking changes), but that looks good to me.

File: docs/sources/administration/service-accounts/migrate-api-keys.md [hunk 1]
```diff
@@ -1,93 +1,37 @@
 ---
 aliases:
-  - about-api-keys/
-  - create-api-key/
-description: This section contains information about API keys in Grafana
+  - ../api-keys/ # /docs/grafana/<GRAFANA_VERSION>/administration/api-keys/
+  - ../about-api-keys/ # /docs/grafana/<GRAFANA_VERSION>/administration/about-api-keys/
+  - ../create-api-key/ # /docs/grafana/<GRAFANA_VERSION>/administration/create-api-key/
```
Comment (jdbaldry) [inline docs/sources/administration/service-accounts/migrate-api-keys.md:5]: ❤️

File: docs/sources/administration/service-accounts/migrate-api-keys.md [hunk 2]
```diff
@@ -1,93 +1,37 @@
...
+description: This section contains information about how to migrate legacy API keys to service account tokens
```
Comment (jdbaldry) [inline docs/sources/administration/service-accounts/migrate-api-keys.md:6]: "This section" won't make sense in the contexts where users see the description (search engine results and social media posts)

```suggestion
description: Learn how to migrate legacy API keys to service account tokens.
```

**Review by jdbaldry — APPROVED** [review]
Rest of docs changes LGTM, thanks!

**Comment by eleijonmarck** [comment]
@ppcano would this mean, if you go to the previous API key page in grafana docs, you would get redirected to service accounts?

We are still going to have lots of customers relying on API keys for some time for the previous versions of Grafana. And i want to make sure that that is the case here right? The docs here only updates from the current version.

**Comment by eleijonmarck** [comment]
https://github.com/grafana/grafana/issues/53567

**Review by IevaVasiljeva — APPROVED** [review]
Nice, all looks good to me 👌

**Comment by eleijonmarck** [comment]
@ppcano i'd be hesitant to backport this to v11.1. How come we are backporting this change?

**Comment by ppcano** [comment]
@eleijonmarck I've now removed the v11.1 backport. Should we also remove the v11.2 backport? 

> would this mean, if you go to the previous API key page in grafana docs, you would get redirected to service accounts?

No, users will be redirected to the `Migrate API keys` page. I think for `next`, `latest`, and version backports. 

It is nearly the same content: 
- changed the title 
- nested it under the Service Accounts docs.
- remove the instructions to create API Keys as we discourage them

> We are still going to have lots of customers relying on API keys for some time for the previous versions of Grafana. And i want to make sure that that is the case here right? The docs here only updates from the current version.

The API key docs won't disappear, so it seems a safe change. But please let me know if you think we should do any additional change.

**Comment by eleijonmarck** [comment]
Keep the back port for 11.2 and i think this is a great change. As all redirects or previous links that people have will end up in service accounts. I like it ☀️

**Review by eleijonmarck — APPROVED** [review]

**Review by Eve832 — APPROVED** [review]
Thank you for this PR!! lgtm

**Review by irenerl24 — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**